src/shims/unix/fs.rs

```diff
@@ -1660,6 +1690,45 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn isatty(&mut self, fd: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        #[cfg(unix)]
+        {
+            let fd = this.read_scalar(fd)?.to_i32()?;
```

Comment:
Better don't have two variables called `fd` that have different "types" -- one is a miri FD, one is a host FD.